src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder.cs

```diff
@@ -70,6 +70,20 @@ internal override BoundStatement BindSwitchExpressionAndSections(SwitchStatement
                 locals, functions, switchSections, defaultLabel, this.BreakLabel, this);
         }
 
+        internal override void BindPatternSwitchLabelForInference(CasePatternSwitchLabelSyntax node, DiagnosticBag diagnostics)
+        {
+            // This simulates enough of the normal binding path of a switch statement to cause
+            // the label's pattern variables to have their types inferred, if necessary.
+            BoundPatternSwitchLabel defaultLabel = null;
+            BindPatternSwitchSectionLabel(
```

Comment:
I'll add an assertion.

---
In reply to: [75772273](https://github.com/dotnet/roslyn/pull/13274#discussion_r75772273) [](ancestors = 75772273)